### PR TITLE
Simpler alias elimination

### PR DESCRIPTION
**Pull request by shashi**
as described in #686

### CODE REVIEW

File: src/systems/diffeqs/odesystem.jl [hunk 1]
```diff
@@ -149,3 +149,5 @@ Base.:(==)(sys1::ODESystem, sys2::ODESystem) =
 function rename(sys::ODESystem,name)
     ODESystem(sys.eqs, sys.iv, sys.states, sys.ps, sys.pins, sys.observed, sys.tgrad, sys.jac, sys.Wfact, sys.Wfact_t, name, sys.systems)
 end
+
+isdiffeq(eq) = eq.lhs isa Term && operation(eq.lhs) isa Differential
```
Comment (ChrisRackauckas) [inline src/systems/diffeqs/odesystem.jl:153]: this should go in AbstractODESystem

File: src/systems/reduction.jl [hunk 2]
```diff
@@ -27,59 +29,99 @@ function substitute_aliases(diffeqs, dict)
...
+            return prod(nums), syms[1]
+        end
     else
-        0 ~ eq.lhs - eq.rhs
+        return nothing
     end
 end
 
-function alias_elimination(sys::ODESystem)
+function alias_elimination(sys)
```
Comment (ChrisRackauckas) [inline src/systems/reduction.jl:52]: since the output is an ODESystem, this is currently limited to ODESystem

**Comment by YingboMa** [comment]
This is still not quite right. I will fix it tomorrow.

**Comment by YingboMa** [comment]
I am happy with this PR at its current state. I'll merge once tests pass.

**Pull request merged**